platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/extensions/PhaseTimer.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.metrics.extensions;
+
+import static com.swirlds.common.units.TimeUnit.UNIT_MICROSECONDS;
+import static com.swirlds.common.units.TimeUnit.UNIT_NANOSECONDS;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.metrics.Metrics;
+import com.swirlds.common.metrics.RunningAverageMetric;
+import com.swirlds.common.units.TimeUnit;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Tracks performance metrics for systems that operate in phases.
+ *
+ * @param <T> the type of the phase, must be an enum
+ */
+public class PhaseTimer<T extends Enum<T>> {
+
+    private final Time time;
+
+    private final boolean fractionMetricsEnabled;
+    private final boolean absoluteTimeMetricsEnabled;
+
+    private final Map<T, FractionalTimer> fractionalTimers = new HashMap<>();
```

Comment:
A short explanation of fractional vs absolute timers would be helpful

Reply:
```
    /**
     * Tracks the fraction of time, out of 1.0, spent in each phase.
     */
    private final Map<T, FractionalTimer> fractionalTimers = new HashMap<>();

    /**
     * Tracks the average time spent in a phase before switching to another phase.
     */
    private final Map<T, RunningAverageMetric> absoluteTimeMetrics = new HashMap<>();
```